packages/algorand-js/src/interpreter/ssc.ts

```diff
@@ -0,0 +1,150 @@
+/* eslint sonarjs/no-duplicate-string: 0 */
+import { toBytes } from "algob";
+import { AccountState, AppLocalState, SSCSchemaConfig, SSCStateSchema } from "algosdk";
+
+import { TealError } from "../errors/errors";
+import { ERRORS } from "../errors/errors-list";
+import { compareArray } from "../lib/helpers";
+import { StackElem } from "../types";
+import { Interpreter } from "./interpreter";
+
+// returns new key value pair by setting type and corresponding values
+function getKeyValPair (key: Uint8Array, value: StackElem): SSCStateSchema {
+  let val;
+  if (typeof value !== "bigint") {
+    val = { type: 1, bytes: value, uint: 0 };
+  } else {
+    val = { type: 2, bytes: toBytes(''), uint: Number(value) };
+  }
+
+  return {
+    key: key,
+    value: val
+  };
+}
+
+/**
+ * Description: assert if the given key-value pairs are valid by schema
+ * @param keyValue: list of key-value pairs (state data)
+ * @param schema: permissible local/global state schema
+ */
+function assertValidSchema (keyValue: SSCStateSchema[], schema: SSCSchemaConfig): void {
+  let numUint = 0; let byteSlices = 0;
+  for (const k of keyValue) {
+    k.value.type === 1 ? byteSlices++ : numUint++;
+  }
+  if (numUint > schema["num-uint"] || byteSlices > schema["num-byte-slice"]) {
+    throw new TealError(ERRORS.TEAL.INVALID_SCHEMA);
+  }
+}
+
+/**
+ * Description: fetches local state value for key present in account
+ * returns undefined otherwise
+ * @param appId: current application id
+ * @param account: account having local state data
+ * @param key: key to fetch value of from local state
+ */
+export function getLocalState (appId: number, account: AccountState, key: Uint8Array): StackElem | undefined {
+  const localState = account["apps-local-state"];
+  const data = localState.find(state => state.id === appId)?.["key-value"]; // can be undefined (eg. app opted in)
+  if (data) {
+    const keyValue = data.find(schema => compareArray(schema.key as Uint8Array, key));
+    const value = keyValue?.value;
+    if (value) {
+      return value?.bytes as Uint8Array || BigInt(value?.uint);
+    }
+  }
+  return undefined;
+}
```

Comment:
this should be a method of the `AccountState` class.

Reply:
Sorry, i don't follow. Did you mean the `AccountState` interface? If we add this method to the interface, then we will defer from algosdk's representation of `Account Information`. ( `getLocalState` is a custom function)

Or did you mean the `MockAccount` class which we will be adding in PoC? Should this method be moved there?
Or something else?

Reply:
I have added subtask for moving this to the `MockAccount` class in PoC